sdk/core/azure-core/inc/context.hpp

```diff
@@ -39,6 +39,13 @@ namespace Azure { namespace Core {
     };
 
   public:
+
+// The compiler gives a false positive on the union type, disable that warning for this constructor
+#if _MSC_VER
```

Comment:
I suggest `#ifdef _MSC_VER` instead of `#if _MSC_VER`: some compilers at some warning levels will warn for the latter when `_MSC_VER` is not defined despite the behavior being well-defined. (Also on line 87.)

Reply:
Agree! I think we now even spell out all `#ifdef`s as `#if defined(...)` everywhere in our code

Reply:
Ah, that shouldn't be the case anymore, sorry for necroposting!